tests/feature/feature_console/deregister.pm

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2015-2017 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use registration;
+
+sub run() {
+    select_console 'root-console';
+
+    # check and list registered modules
+    my $out = script_output "SUSEConnect --status-text | tee /tmp/registration_status";
+    $out =~ /Registered/ or die "There's no module registered";
+    my $count = script_output "grep -c \"Status: ACTIVE\" /tmp/registration_status";
+
+    # check if a certain module is registered, e.g. containers, which has a regcode
+    # alternative: check all registered modules in the same way?
+    # FIXME my $all_reg_modules = script_output 'grep -B 3 '^\s*Registered' /tmp/registration_status';
+    my $module = "containers";
+    assert_script_run "grep -A 3 $module /tmp/registration_status | grep \'^\s*Registered\'";
+
+    # deregister the module and check if it is successful
+    assert_script_run "SUSEConnect --de-register $module";
```

Comment:
-d, --de-register        De-registers the system and base product, or in
                             conjunction with --product, a single extension, and
                             removes all its services installed by SUSEConnect.
                             After de-registration the system no longer consumes
                             a subscription slot in SCC.


so current code deregister whole system